api/vulnerability_exceptions.go

```diff
@@ -0,0 +1,448 @@
+//
+//   client.V2.VulnerabilityExceptions.Create(vulnerabilityException)
+//
+func NewVulnerabilityException(name string, exception VulnerabilityExceptionConfig) VulnerabilityException {
+	packages := aggregatePackages(exception.Package)
+	vulnException := VulnerabilityException{
+		Enabled:         1,
+		ExceptionName:   name,
+		ExceptionReason: exception.ExceptionReason.String(),
+		Props:           VulnerabilityExceptionProps{Description: exception.Description},
+		VulnerabilityCriteria: VulnerabilityExceptionCriteria{
+			Severity: exception.Severities.ToStringSlice(),
+			Package:  packages,
+			Cve:      exception.Cve,
+			Fixable:  exception.FixableEnabled(),
+		},
+		ExpiryTime: exception.ExpiryTime.UTC().Format(lwtime.RFC3339Milli),
+	}
+	vulnException.setResourceScope(exception.ResourceScope)
+	return vulnException
+}
+
+func aggregatePackages(packages []VulnerabilityExceptionPackage) []map[string][]string {
+	var packs []map[string][]string
+	for _, pck := range packages {
+		var packagesMap = make(map[string][]string)
+		//aggregate packages with same name
+		if len(packs) > 0 {
+			if _, ok := packs[0][pck.Name]; ok {
+				packs[0][pck.Name] = append(packs[0][pck.Name], pck.Version)
+				continue
+			}
+		}
+		packagesMap[pck.Name] = []string{pck.Version}
+		packs = append(packs, packagesMap)
+	}
+	return packs
+}
+
+func (exception *VulnerabilityException) setResourceScope(scope vulnerabilityExceptionResourceScope) {
+	switch scope.Type() {
+	case VulnerabilityExceptionTypeContainer:
+		ctr := scope.Scope()
+		exception.ExceptionType = VulnerabilityExceptionTypeContainer.String()
+		exception.ResourceScope = VulnerabilityExceptionResourceScope{
+			ImageID:    ctr.ImageID,
+			ImageTag:   ctr.ImageTag,
+			Registry:   ctr.Registry,
+			Repository: ctr.Repository,
+			Namespace:  ctr.Namespace,
+		}
+	case VulnerabilityExceptionTypeHost:
+		host := scope.Scope()
+		exception.ExceptionType = VulnerabilityExceptionTypeHost.String()
+		exception.ResourceScope = VulnerabilityExceptionResourceScope{
+			Hostname:    host.Hostname,
+			ClusterName: host.ClusterName,
+			ExternalIP:  host.ExternalIP,
+			Namespace:   host.Namespace,
+		}
+	default:
+		exception.ResourceScope = VulnerabilityExceptionResourceScope{}
+	}
+}
+
+func (exception VulnerabilityException) Status() string {
+	if exception.Enabled == 1 {
+		return "Enabled"
+	}
+	return "Disabled"
+}
+
+func (cfg VulnerabilityExceptionConfig) FixableEnabled() []int {
+	if cfg.Fixable {
+		return []int{1}
+	}
+	return []int{0}
+}
+
+// List returns a list of Vulnerability Exceptions
+func (svc *VulnerabilityExceptionsService) List() (response VulnerabilityExceptionsResponse, err error) {
+	err = svc.client.RequestDecoder("GET", apiV2VulnerabilityExceptions, nil, &response)
+	return
+}
+
+// Create creates a single Vulnerability Exception
+func (svc *VulnerabilityExceptionsService) Create(vuln VulnerabilityException) (
```

Comment:
I would suggest removing `Create()` and `Update()`  functions since we have specific functions for every type and that way we will follow a pattern.